share/functions/edit_command_buffer.fish

```diff
@@ -0,0 +1,45 @@
+function edit_command_buffer --description 'Edit the command buffer in an external editor'
+    set -l f (mktemp)
+    if set -q f[1]
+        mv $f $f.fish
```

Comment:
I think you know that this is racy. Though it seems a rather theoretical criticism since this function won't be executed in a tight loop by multiple shells at the same time.

Any fix to this would require either an alternative to mktemp or a mktemp that accepts a suffix, which is unlikely to be available on BSD or macOS.